### PR TITLE
refactor(contracts): separate v4/v5 init slot clearing in L2ContractsManagerUtils

### PR DESCRIPTION
**Pull request by maurelian**
## Summary
- Separate the interleaved v4/v5 initialization slot clearing into distinct code blocks — v4 path is fully contained in an `if (_slot != v5Slot)` block, v5 path runs unconditionally below
- Add `L2ContractsManager_InvalidV5Offset` revert when v5 slot is passed with a non-zero offset
- Add tests for non-zero `_offset` (previously untested) and the new v5 offset validation
- Bump L2ContractsManager to 1.4.0

## Motivation
The previous code always ran both v4 and v5 clearing logic, relying on implicit no-ops for the version that doesn't apply. When `_slot == v5Slot`, this caused a redundant SSTORE and a confusing re-read of a slot that was just written to. The refactored version is semantically identical but easier to reason about.

## Test plan
- [x] All 9 `L2ContractsManagerUtils_UpgradeToAndCall_Test` tests pass
- [x] New: `test_upgrade_v4NonZeroOffset_succeeds` — exercises offset=2 bit-shifting
- [x] New: `test_upgrade_v4InitializingNonZeroOffset_reverts` — revert with offset=2
- [x] New: `test_upgrade_v5InvalidOffset_reverts` — validates new offset guard

### CODE REVIEW

File: packages/contracts-bedrock/test/libraries/L2ContractsManagerUtils.t.sol [hunk 1]
```diff
@@ -192,6 +192,81 @@ contract L2ContractsManagerUtils_UpgradeToAndCall_Test is CommonTest {
...
+            implV2,
+            _storageSetterImpl,
+            abi.encodeCall(L2ContractsManagerUtils_ImplV2_Harness.initialize, ()),
+            INITIALIZABLE_SLOT_OZ_V4,
+            offset
+        );
+    }
+
+    /// @notice Tests that upgrade reverts when v5 slot is passed with a non-zero offset.
+    function test_upgrade_v5InvalidOffset_reverts() public {
```
Comment (maurelian) [inline packages/contracts-bedrock/test/libraries/L2ContractsManagerUtils.t.sol]: make this a fuzz test.

**Review by 0xOneTony — APPROVED** [review]
lgtm

**Review by 0xiamflux — APPROVED** [review]

File: packages/contracts-bedrock/src/L2/L2ContractsManager.sol [hunk 2]
```diff
@@ -35,8 +35,8 @@ contract L2ContractsManager is ISemver {
     error L2ContractsManager_FeatureFlagMismatch();
 
     /// @notice The semantic version of the L2ContractsManager contract.
-    /// @custom:semver 1.3.0
-    string public constant version = "1.3.0";
+    /// @custom:semver 1.4.0
+    string public constant version = "1.4.0";
```
[the remark this answers is not in the document]
Comment (maurelian) [reply packages/contracts-bedrock/src/L2/L2ContractsManager.sol:39]: just a PR description issue, not helpful.

**Review by stevennevins — APPROVED** [review]
